runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/DispatchAgent.java

```diff
@@ -1311,9 +1349,16 @@ private MessageConsumer handleBeginInitial(
             if (newStream != null)
             {
                 final long replyId = supplyReplyId(initialId);
-                streams[streamIndex(initialId)].put(instanceId(initialId), newStream);
-                throttles[throttleIndex(replyId)].put(instanceId(replyId), newStream);
+                final int streamIndex = streamIndex(initialId);
+                final int throttleIndex = throttleIndex(replyId);
+                streams[streamIndex].put(instanceId(initialId), newStream);
+                throttles[throttleIndex].put(instanceId(replyId), newStream);
```

Comment:
Look for calls to `streams[streamIndex].remove(...)` and `throttles[streamIndex].remove(...)` too, so we can cleanup binding streams and throttles properly too.